apps/hannk/interpreter/tensor.cpp

```diff
@@ -192,13 +222,20 @@ void Tensor::set_alias_of(const TensorPtr &t, const SmallVector<int, max_rank> &
     }
     auto &shared_buffer = storage_->buffer;
     assert(shared_buffer.type().bytes() == type().bytes());
-    assert(shared_buffer.dimensions() == (int)offset_bounds.size());
     assert(!shared_buffer.data());
 
     // Check that the storage is big enough for this buffer.
-    for (int i = 0; i < shared_buffer.dimensions(); i++) {
-        assert(offset_bounds[i].min >= shared_buffer.dim(i).min());
-        assert(offset_bounds[i].max <= shared_buffer.dim(i).max());
+    if (this->rank() == t->rank()) {
```

Comment:
I think this is a little sketchy. I think reshapes of the same rank are legal, where each dimension has a different extent, but the total number of elements is the same. I think this either needs to be a separate function, or a parameter indicating the aliasing should share "flat" indices or "rank-dimensional" indices.

Reply:
Originally I just had a flag inside the Tensor indicating whether the aliasing was for a reshape or not, but that felt clumsy -- I didn't think about the same-rank, different-extent case though. I'll revisit this.